__tests__/commands/install/workspaces-install.js

```diff
@@ -69,6 +69,24 @@ test.concurrent("workspaces warn and get ignored if they don't have a name and a
   );
 });
 
+test.concurrent('should not warn unmet peer for satisfied transient peer dependency on workspace package', (): Promise<
+  void,
+> => {
+  return buildRun(
+    reporters.BufferReporter,
+    path.join(__dirname, '..', '..', 'fixtures', 'install'),
+    async (args, flags, config, reporter, lockfile): Promise<void> => {
+      const install = new Install(flags, config, reporter, lockfile);
+      await install.init();
+      const warnings = reporter.getBuffer().filter(({type}) => type === 'warning').map(({data}) => data);
+      expect(warnings).toEqual(['" > nomore-test-b@0.0.1" has unmet peer dependency "nomore-test-a@^0.0.1".']);
```

Comment:
It's a little weird that the test is named "should not warn" then asserts that there _is_ a warning.
I'm assuming this warning that still shows up in expected and the test assumption is that another one isn't showing?

Maybe change the expectation to read the same as the test name; expect that a certain warning isn't in the collection?

Reply:
I rephrased the name of the test to `should only warn for unmet peer dependency on workspace package`.
Actually if the test was failing (e.g. if I keep the old code running) there would be additional unmet peer warnings which would just be wrong. So what I wanted to test here was that there would be only one warning emitted and it would be that one.
The issue I have is that testing for a specific warning message not to appear sounded like an invite for falsely passing tests which are really awkward for this kind of feature to troubleshoot "after the battle".